### DISCUSSION TITLE
x86Duino (Embedded X86DX2);8086/80286/80386/80486 Support?

### DISCUSSION THREAD

**Discussion by hilga007** [chettrick/discobsd#16] [Ideas]
Hello! DM&P of Taiwan makes a lot of their own Vortex86 and 86Duino products, using real x86 architecture. I have spent a long time fretting about getting a BSD on there for a UNIX-like experience and here we are! 

What would it take to have an x86Duino version? (Bonus points for magically supporting the Intel neural compute stick as well) :3 

https://www.86duino.com/

300MHz, 1GB DDR3, 128MB flash memory + DiscoBSD = One HELL of a party! https://www.86duino.com/?p=68

(If you do this, could you support i486 as well? Not all i586/i686 software will work with some VorteX86 CPUs due to lack of the CMOV instruction. 

----

Also: New-Vintage and Neo-Retro boards and devices are continuing to be released with 8086, 80286, 80386 CPUs. 

i386 to run on these new tiny handhelds would be lovely... 

Feasibility I, the humble fan and user and nerd, leave up to you! 

---- 

8086/80286. This goes back to the 2.11BSD part (which has 16-bit versions) and seeing which pieces of the other puzzles will backport to 16-bit. 

Would be cool to see a tiny DiscoBSD/16 up against ELKS for some non-proprietary UNIX choice at the tiniest of levels.

**Comment by chettrick** [comment]
Hey there. 

> Hello! DM&P of Taiwan makes a lot of their own Vortex86 and 86Duino products, using real x86 architecture. I have spent a long time fretting about getting a BSD on there for a UNIX-like experience and here we are!

Have you tried out NetBSD or OpenBSD on the 86Duino boards? They both have supported the Vortex86EX in the past, but I think a current OpenBSD release (OpenBSD 6.8 or newer) requires at least an i586, although OpenBSD 5.4 ran on an 86Duino EduCake. NetBSD can run on almost anything. Check these dmesgs:

https://dmesgd.nycbug.org/index.cgi?fts=vortex86ex 

Another option would be to port it to [LiteBSD](https://github.com/sergev/litebsd), but I think NetBSD has the greatest chance of working out of the box, and with all the luxuries of their base system and pkgsrc.

> What would it take to have an x86Duino version?

DiscoBSD on an x86 is wholly mismatched. A specific constraint of DiscoBSD, inherited from RetroBSD and 2.11BSD, is that it does not do virtual memory (no memory management unit). The overlays feature of 2.11BSD was also removed in RetroBSD, so DiscoBSD does not currently do overlays either.

An 8088/8086 in real mode would be an interesting porting experience, though.

Do you have any of the 86Duino hardware? If so, try out [NetBSD/i386](https://wiki.netbsd.org/ports/i386) and let us know how it goes!

**Reply by hilga007** [reply]
Thanks for the thoughtful and in-depth reply. You are truly phenomenal and I appreciate you! I have some 86duino stuff in my tinker drawer along with Galileo boards and neural compute sticks xD

I also have some Vortex86DX2 thin  clients. I'll let ya know how NetBSD/i386 works out! 

The 8008, 8080, 8085, 8088, 8086 would all be interesting. 

The 8008 has 12-bit address space, I believe, with an 8-bit data bus. 
The 8080, 8085, 8088/V20 have 16-bit address space with 8-bit data bus. 
The 8086/V30 have 16-bit address space with 16-bit data bus but no real mode. 

Would any of those fellas be  viables candidate for your Disco'd Up BSD?!

Love to see your work out here regardless. This is so very super cooll.
